README.adoc

```diff
@@ -17,6 +17,34 @@ An end-to-end https://www.optaplanner.org/[OptaPlanner] web application ready to
 
 See https://www.youtube.com/watch?v=sOWC4qrXxFk[the video demo].
 
+== Quick Start
+
+Install Docker and run the following on the command line:
+
+[source,shell]
+----
+docker build -t rostering .
+docker run -p 8080:8080 --rm -it -e SPRING_PROFILES_ACTIVE=dev rostering
+----
+This takes a few minutes on the first time. The application will be automatically populated with test data.
+
+You can then access the application on localhost:8080, once it finishes starting
+
+== Development
+
+=== Backend
+
+Configure your IDE to run the main class, which is in
+`org.optaweb.employeerostering.OptaWebEmployeeRosteringApplication` in `employee-rostering-backend` repository.
+
+You will need to specify this program argument for Spring Boot to run in development mode and provision the H2 in-memory database:
+
+`--spring.profiles.active=dev`
```

Comment:
Side discussion: I'm starting to think that configuring persistence for PostgreSQL in the default (production) profile was a mistake. My thinking was that production is the app's ultimate purpose so default configuration must target production (and file-based H2 database is certainly not meant for production, while PostgreSQL is). But now I realize that the app is being run 99% in development mode and having to provide `--spring.profiles.active=dev` every time doesn't make life much easier. So I'm thinking about moving file-based H2 configuration to the default (production) profile (see the contradiction?) and PostgreSQL to a postgresql profile. Haven't decided yet. What are your opinions guys, @Christopher-Chianelli, @cuijulian, @rsynek, @ge0ffrey.

Reply:
Could you put `spring.profiles.active=dev` in `application.properties` to achieve the same effect (using `-Dspring.profiles.active=production` should override it)?

Reply:
@yurloc I agree with your point, it makes sense that most users try out the application with an in-memory db first, rather than production. I also think that using H2 as the default makes sense (ie. using the `dev` profile as the default), so @Christopher-Chianelli's suggestion makes sense.

Reply:
" that the app is being run 99% in development mode and having to provide --spring.profiles.active=dev every time doesn't make life much easier." Absolutely. It increased the hurdle to get started. I experienced it on Friday - the lack of spring profile dev - and I'd really like to see the readme with a short, simple command that just builds and runs the app locally (without a need to install docker etc).

Reply:
@yurloc that's a good point, the getting started should be as simple as possible.